op-node/p2p/peer_scorer.go

```diff
@@ -1,16 +1,99 @@
 package p2p
 
 import (
+	"fmt"
+	"sort"
+	"strconv"
+	"strings"
+
 	log "github.com/ethereum/go-ethereum/log"
 	pubsub "github.com/libp2p/go-libp2p-pubsub"
 	peer "github.com/libp2p/go-libp2p/core/peer"
 )
 
 type scorer struct {
-	peerStore Peerstore
-	metricer  GossipMetricer
-	log       log.Logger
-	gater     PeerGater
+	peerStore           Peerstore
+	metricer            GossipMetricer
+	log                 log.Logger
+	gater               PeerGater
+	bandScoreThresholds BandScorer
+}
+
+// scorePair holds a band and its corresponding threshold.
+type scorePair struct {
+	band      string
+	threshold float64
+}
+
+// BandScoreThresholds holds the thresholds for classifying peers
+// into different score bands.
+type BandScoreThresholds struct {
+	bands []scorePair
+}
+
+// BandScorer is an interface for placing peer scores
+// into various bands.
+//
+// Implementations are expected to construct internals using the
+// [Parse] function and then expose the [Bucket] function for
+// downstream [BandScorer] consumers.
+type BandScorer interface {
+	Bucket(score float64) string
+	Reset()
```

Comment:
remove

Reply:
alrighty...

Reply:
yes, if there's a reason for an inteface

Reply:
The other half to `accept interfaces, return structs` is that the interface should (usually) be defined at the point of usage, not in the code which defines the struct (unless it's something like `io.Read` where there are a lot of different implementations in the same package)